moveit_ros/planning/plan_execution/src/plan_execution.cpp

```diff
@@ -271,8 +271,7 @@ bool plan_execution::PlanExecution::isRemainingPathValid(const ExecutableMotionP
 bool plan_execution::PlanExecution::isRemainingPathValid(const ExecutableMotionPlan& plan,
                                                          const std::pair<int, int>& path_segment)
 {
-  if (path_segment.first >= 0 && path_segment.second >= 0 &&
-      plan.plan_components_[path_segment.first].trajectory_monitoring_)
+  if (path_segment.first >= 0 && plan.plan_components_[path_segment.first].trajectory_monitoring_)
```

Comment:
Can you add a comment explaining that, if ``path_segment.second == -1`` then this function will fall back to collision checking the entire trajectory?